WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/InPersonPaymentsMenuViewController.swift

```diff
@@ -334,6 +339,67 @@ private extension InPersonPaymentsMenuViewController {
             self.navigationController?.show(connectionController, sender: nil)
         }.store(in: &cancellables)
     }
+
+    private func createFeedbackBanner() -> TopBannerView {
+        let giveFeedbackAction = TopBannerViewModel.ActionButton(
+            title: Localization.giveFeedbackButton,
+            action: { _ in
+                self.displayFeedbackBannerSurvey()
+            })
+        let dismissFeedbackAction = TopBannerViewModel.ActionButton(
+            title: Localization.dismissFeedbackButton,
+            action: { _ in
+                self.dismissFeedbackBanner()
+            })
+        let viewModel = TopBannerViewModel(
+            title: Localization.feedbackBannerTitle,
+            infoText: Localization.feedbackBannerContent,
+            icon: .speakerIcon.withRenderingMode(.alwaysTemplate),
+            iconTintColor: .wooCommercePurple(.shade50),
+            isExpanded: false,
+            topButton: .chevron(handler: {
+                self.tableView.updateHeaderHeight()
+            }),
+            actionButtons: [giveFeedbackAction, dismissFeedbackAction]
+        )
+        let topBannerView = TopBannerView(viewModel: viewModel)
+        topBannerView.translatesAutoresizingMaskIntoConstraints = false
+        return topBannerView
+    }
+
+    private func configureFeedbackBanner() {
+        topBannerView = createFeedbackBanner()
+        //tableView.updateHeaderHeight()
+        showFeedbackBanner()
+    }
+
+    private func showFeedbackBanner() {
+        guard let topBannerView = topBannerView else { return }
+
+        // Configures header container view:
+        // TODO: Find the correct width and height
+        let width = Int(tableView.frame.width)
+        let height = 100
```

Comment:
At the moment I'm hardcoding the banner height here as may change with the different banner content once we show different surveys. Will address this bit on a future PR, but happy to get any recommendation on how to deal with this on a better way.

Reply:
You can take a look at `CouponsListViewController` to check how the banner is added. If I see that correctly, we don't even need to specify a hardcoded height; it is rendered automatically thanks to the `updateHeaderHeight`  method in our `UITableView+HeaderFooterHelpers` extension.
I recommend looking at the used approach there to add the banner, not only for the height but in general to keep consistency between the two implementations.